modules/ROOT/pages/configure-horizontal-autoscaling.adoc

```diff
@@ -0,0 +1,200 @@
+
+
+
+[NOTE]
+Refer to your managed vendor documentation for cluster autoscaling best practices and over-provisioning configuration details.
+
+
+=== Configure Topology Spread Constraints
+
+Your Mule app deployments which are enabled with autoscaling, are required to customize the kubernetes resources for the topology spread constraints in pod spec by updating the *whenUnsatisfiable* field with `DoNotSchedule`.
+This configuration ensures that autoscaling replicas are evenly spread across zones for high availability.
+
+----
+topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: DoNotSchedule
+    labelSelector:
+      matchLabels:
+        rtf.mulesoft.com/id: {{ .Values.id }}
+----
+
+For further details on the configuration, refer to xref:customize-kubernetes-crd.adoc[] documentation.
+
+[WARNING]
+Customizing topology spread with `whenUnsatisfiable: DoNotSchedule`, without the recommended configurations for node groups per availability zone, cluster autoscaler, and overprovisionining, can lead to K8s failing to schedule the replicas with `Pending state` error.
+
+=== Understand Autoscaling Policy
+
+MuleSoft owns and applies the https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/[autoscaling^] policy for your Mule application deployments.
+
+The CPU based HPA policy used for all Mule apps deployed on Runtime Fabric instances, is as follows:
+
+----
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: my-app
+  namespace: app-namespace
+spec:
+  behavior:
+    scaleDown:
+      policies:
+      - periodSeconds: 15
+        type: Percent
+        value: 100
+      selectPolicy: Max
+      stabilizationWindowSeconds: 300
+    scaleUp:
+      policies:
+      - periodSeconds: 180
+        type: Percent
+        value: 100
+      selectPolicy: Max
+      stabilizationWindowSeconds: 0
+  maxReplicas: 3
+  metrics:
+  - resource:
+      name: cpu
+      target:
+        averageUtilization: 70
+        type: Utilization
+    type: Resource
+  minReplicas: 1
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: my-app
+
+----
+
+Some points to consider:
+
+* Up to 100% running pods may be added until the maximum configured replicas is reached.
+* The number of pods added is based on the aggregated calculations over the past 180 seconds.
+* The number of pods removed is based on the aggregated calculations over the past 300 seconds.
+* Max scale up profile is 1 -> 2 -> 4 -> 8, where MuleSoft hits 8 replicas in approximately 6 minutes.
+
+The following are few examples of Mule app types that fit CPU based HPA:
+
+* Mule apps that scale based on CPU usage only.
+* HTTP/HTTPS apps with async requests.
+
+
+== Performance Considerations and Limitations
+
+In Runtime Fabric, the policy in use was benchmarked for Mule apps with CPU Reserved: 0.45vCpu and Limit: 0.55vCpu, which corresponds to these settings:
```

Comment:
We should also add:
Scale up and scale down performance will vary based on the replica size and the application type.

Reply:
This section should also include autoscaling performance -- Avg time to spin up new replicas

Reply:
@rithishapadmanabh could you provide content on _autoscaling performance -- Avg time to spin up new replicas?_